x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/service/ElasticServiceAccounts.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.security.authc.service;
+
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.xpack.core.security.authz.RoleDescriptor;
+import org.elasticsearch.xpack.core.security.user.User;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+final class ElasticServiceAccounts {
+
+    static final String NAMESPACE = "elastic";
+
+    private static final ServiceAccount FLEET_ACCOUNT = new ElasticServiceAccount("fleet",
+        new RoleDescriptor(
+            NAMESPACE + "/fleet",
+            new String[]{"monitor", "manage_own_api_key"},
+            new RoleDescriptor.IndicesPrivileges[]{
+                RoleDescriptor.IndicesPrivileges
+                    .builder()
+                    .indices("logs-*", "metrics-*", "traces-*")
+                    .privileges("write", "create_index", "auto_configure")
+                    .build()
+            },
+            null,
+            null,
+            null,
+            null,
+            null
+        ));
```

Comment:
I think this descriptor is good for now, but we'll need to check it with the Fleet team later on.
It's not quite identical to the `fleet_enroll` role that that Fleet creates today, and we'll want to work through the differences.